x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rule_details/components/view_in_app.test.tsx

```diff
@@ -7,69 +7,91 @@
 
 import * as React from 'react';
 import { v4 as uuidv4 } from 'uuid';
-import { ReactWrapper } from 'enzyme';
-import { act } from 'react-dom/test-utils';
-import { mountWithIntl } from '@kbn/test-jest-helpers';
+import { render, screen, waitFor } from '@testing-library/react';
+import { __IntlProvider as IntlProvider } from '@kbn/i18n-react';
 
 import { Rule } from '../../../../types';
 import { ViewInApp } from './view_in_app';
 import { useKibana } from '../../../../common/lib/kibana';
+
+const mockUseKibana = useKibana as jest.MockedFunction<typeof useKibana>;
+
 jest.mock('../../../../common/lib/kibana');
 
 jest.mock('../../../lib/capabilities', () => ({
   hasSaveRulesCapability: jest.fn(() => true),
 }));
 
+const renderWithIntl = (ui: React.ReactElement) => {
+  return render(
+    <IntlProvider locale="en" messages={{}}>
+      {ui}
+    </IntlProvider>
+  );
+};
+
 describe('view in app', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+
+    mockUseKibana.mockReturnValue({
+      services: {
+        application: {
+          navigateToUrl: jest.fn(),
+          navigateToApp: jest.fn(),
+        },
+        http: {
+          basePath: {
+            prepend: jest.fn((path: string) => path),
+          },
+        },
+        alerting: {
+          getNavigation: jest.fn(),
+        },
+      },
+    } as unknown as ReturnType<typeof useKibana>);
+  });
+
   describe('link to the app that created the rule', () => {
     it('is disabled when there is no navigation', async () => {
       const rule = mockRule();
-      const { alerting } = useKibana().services;
-      let component: ReactWrapper;
-      await act(async () => {
-        // use mount as we need useEffect to run
-        component = mountWithIntl(<ViewInApp rule={rule} />);
+      const alerting = mockUseKibana().services.alerting;
```

Comment:
I find it weird that we mockUserKibana in here and in `beforeEach`.

This test and the next could be simplified if the class defined `const getNavigation = jest.fn();` and passed it in line 49.

Then this test would do

```
getNavigation.mockResolvedValueOnce(null);
...
expect(getNavigation).toBeCalledWith(rule.id);
```

The next test would do `getNavigation.mockResolvedValue('/rule');` instead.